### PR TITLE
Changes to nightmares interactions with flashes and flash bangs.

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request

Nightmares now take a large amount of burn damage from flashes and are immune from the stun.
Flashbangs now do minor amounts of damage at a distance but a shit ton if on the same tile.

## Why It's Good For The Game

This is mostly flavor stuff but the main reason I did this is because of the insane appeal sunglasses have to a nightmare. Flashes do jack shit to a nightmare with sunglasses or other flash protection and in a world where you can run to a nearby patch of darkness before the baton knocks you over you are pretty invincible.

## Changelog
:cl:
tweak: Nightmares now take damage from sources of concentrated light like flashbangs and flashes.
tweak: Nightmares are no longer stunned by flashes.
/:cl:

### CODE REVIEW

**Comment by tmtmtl30** [comment]
i disagree, nightmares are usually pretty weak. it's surprisingly hard to counter being mobbed by three assistants with flashlights and stunbatons in maint

**Comment by tralezab** [comment]
agree with tmt but i'd say that allowing the nightmare to become immune to the one thing you would think to use against a nightmare is bad design so this is a good idea

**Comment by necromanceranne** [comment]
I like this, it's actually quite thematic for nightmares to be vulnerable to these things in a unique way.

File: code/game/objects/items/grenades/flashbang.dm [hunk 1]
```diff
@@ -34,8 +34,12 @@
 		M.Paralyze(20)
 		M.Knockdown(200)
 		M.soundbang_act(1, 200, 10, 15)
+		if HAS_TRAIT(M, TRAIT_FLASHBURN)
+			M.adjustFireLoss (150) //no coming back from that
```
Comment (ATH1909) [inline code/game/objects/items/grenades/flashbang.dm]: I mean, I think they technically could if they get left in darkness for a while (I think that that's fine, I'm just nitpicking the COMMENT here, not the thing being commented on).

File: code/modules/assembly/flash.dm [hunk 2]
```diff
@@ -143,6 +143,8 @@
 		return FALSE
 	if(iscarbon(M))
 		flash_carbon(M, user, 5, 1)
+		if HAS_TRAIT(M, TRAIT_FLASHBURN)
+			M.adjustFireLoss (20)
```
Comment (ATH1909) [inline code/modules/assembly/flash.dm]: this feels a bit low; maybe up the damage to 25 or 30? Keep in mind that a lit welder will usually do 15 damage to a nightmare, due to their inability to wear most kinds of armor.

**Comment by ATH1909** [comment]
I also feel like this trait should have at least SOME interaction with using a flash in hand (to produce an AoE flash effect) or autoflashers, but I feel like that would lead to cheese, since you can hook handheld flashes up to signalers to activate multiple AoE flashes with one signaler pulse, potentially one-shotting a nightmare even if AoE flashes only did 5 damage to them.

**Comment by Tlaltecuhtli** [comment]
you can literally grab a flashbang box, rig it with a voice actived firecracker and then nuke nightmares

File: code/game/objects/items/grenades/flashbang.dm (hunk 1 above)
Comment (itseasytosee) [reply code/game/objects/items/grenades/flashbang.dm]: Not sure if this alpys but they have is a 1.5 modifier for burn but that may just be for weapons

**Comment by itseasytosee** [comment]
As a being of pure darkness would 8 flashbangs going off at the same time not burn away your very being?

**Comment by SkeletalElite** [comment]
> As a being of pure darkness would 8 flashbangs going off at the same time not burn away your very being?

sure but if the meta the second there is a nightmare is to just insta nuke them with flashbangs which has literally no downside that doesn't create very good gameplay. It'd be like suicide maxcaping any other antag except the maxcap does not damage to the station, and doesn't kill anyone except the antag.

**Comment by ExcessiveUseOfCobblestone** [comment]
the flash going from literally useless if wearing glasses to 1hitcrit seems like a very steep change.

**Comment by itseasytosee** [comment]
It is in no way a 1 hit crit it doses around 20 burn

**Comment by ExcessiveUseOfCobblestone** [comment]
oh that's the flashbang... even more lovely lol

**Comment by itseasytosee** [comment]
That's the flashbang if put directly on top of or in.

**Comment by Shadowflame909** [comment]
Honestly the only balance change nightmares need is being able to wear every clothing item. But dropping them all upon jaunting into a being of pure darkness.

Darkness can't wear clothes!

**Comment by itseasytosee** [comment]
go back to #49324

**Comment by SkeletalElite** [comment]
I think that it would be best if you could only be affected by the flashbang damage once every few seconds that way if someone detonates like 8 at the same time it only does the damage of one of them.

**Comment by itseasytosee** [comment]
I could take on a tiny cooldown

File: code/modules/assembly/flash.dm [hunk 3]
```diff
@@ -143,6 +143,8 @@
 		return FALSE
 	if(iscarbon(M))
 		flash_carbon(M, user, 5, 1)
+		if HAS_TRAIT(M, TRAIT_FLASHBURN)
```
Comment (ShizCalev) [inline code/modules/assembly/flash.dm]: add some brackets here please.

File: code/game/objects/items/grenades/flashbang.dm [hunk 4]
```diff
@@ -34,8 +34,12 @@
 		M.Paralyze(20)
 		M.Knockdown(200)
 		M.soundbang_act(1, 200, 10, 15)
+		if HAS_TRAIT(M, TRAIT_FLASHBURN)
+			M.adjustFireLoss (150) //no coming back from that
 	else
 		if(distance <= 1) // Adds more stun as to not prime n' pull (#45381)
 			M.Paralyze(5)
 			M.Knockdown(30)
+			if HAS_TRAIT(M, TRAIT_FLASHBURN)
```
Comment (ShizCalev) [inline code/game/objects/items/grenades/flashbang.dm]: brackets

Comment (itseasytosee) [reply code/game/objects/items/grenades/flashbang.dm]: Will do thanks

**Comment by IndieanaJones** [comment]
Why was this closed?

**Pull request closed without merging**